core/assertion.ts

```diff
@@ -68,7 +69,9 @@ export class Assertion {
 
   public schema(schema: string) {
     this.proto.target.schema = schema;
```

Comment:
i'm a bit concerned about having this block of code repeated in each Action, it seems ripe for something to break. i propose instead:

```
public schema(schema: string) {
  this.session.setNameAndTarget(this.proto, this.proto.target.name, schema);
  return this;
}
```

(i realise that i have also changed it to `return this;`, it's a small bug that we don't have that; i don't mind if you leave it out - we can add it in a later PR)

Reply:
1. The master branch / status quo does not perform that check either. In fact, if you do the suggested change (using `setNameAndTarget`) it crashes with the following error:

      -    {
      -      "fileName": "definitions/override_schema_example.sqlx"
      -      "message": "Duplicate action name detected. Names within a schema must be unique across tables, assertions, and operations: \"override_schema.override_schema_example\""
      -      "stack": "Error: Duplicate action name detected. Names within a schema must be unique across tables, assertions, and operations: \"override_schema.override_schema_example\"\n    at Session.checkTargetIsUnused (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/session.js:303:35)\n    at Session.setNameAndTarget (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/session.js:211:18)\n    at Table.schema (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/table.js:161:26)\n    at Table.config (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/table.js:84:22)\n    at Session.sqlxAction (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/session.js:123:18)\n    at /private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/definitions/override_schema_example.sqlx:21:24\n    at Object.EXTENSIONS.(anonymous function) [as .sqlx] (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/npm/node_modules/vm2/lib/sandbox.js:83:5)\n    at _require (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/npm/node_modules/vm2/lib/sandbox.js:320:24)\n    at Object.<anonymous> (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/index.js:225:13)\n    at NodeVM.run (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/124/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/npm/node_modules/vm2/lib/main.js:448:23)"
      -    }

Same for assertions:

```
           {
      -      "fileName": "definitions/example_assertion.sqlx"
      -      "message": "Duplicate action name detected. Names within a schema must be unique across tables, assertions, and operations: \"hi_there.example_assertion\""
      -      "stack": "Error: Duplicate action name detected. Names within a schema must be unique across tables, assertions, and operations: \"hi_there.example_assertion\"\n    at Session.checkTargetIsUnused (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/session.js:303:35)\n    at Session.setNameAndTarget (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/session.js:211:18)\n    at Assertion.schema (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/assertion.js:61:26)\n    at Assertion.config (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/assertion.js:29:22)\n    at Session.sqlxAction (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/node_modules/@dataform/core/session.js:123:18)\n    at /private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/definitions/example_assertion.sqlx:22:24\n    at Object.EXTENSIONS.(anonymous function) [as .sqlx] (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/npm/node_modules/vm2/lib/sandbox.js:83:5)\n    at _require (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/npm/node_modules/vm2/lib/sandbox.js:320:24)\n    at Object.<anonymous> (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/df/examples/bigquery_language_v2/index.js:33:13)\n    at NodeVM.run (/private/var/tmp/_bazel_cesc/e27ac56fa175c4de03c842b466cd7c26/sandbox/darwin-sandbox/144/execroot/df/bazel-out/darwin-py2-fastbuild/bin/tests/api/api.runfiles/npm/node_modules/vm2/lib/main.js:448:23)"
      -    }
      -    {
```

Since this issue is unrelated to this change, I propose we leave it here for now and file it as an issue.

2. I've changed the function a little bit so that now there is no mention to the suffix in the function. It is all taken care of by `this.session.target(...)`.